code/modules/mob/emote.dm

```diff
@@ -60,20 +60,20 @@
 	var/msg = "<b>[user]</b> " + params
 
 	var/turf/T = get_turf(user) // for pAIs
-	var/broadcast = T ? T : user
-
-	switch (m_type)
-		if (EMOTE_VISIBLE)
-			for(var/mob/O in viewers(broadcast))
-				O.show_message(msg, emote_type)
-			if (!(user in viewers(broadcast)))
-				user.show_message(msg, emote_type)
-
-		if (EMOTE_AUDIBLE)
-			for(var/mob/O in hearers(broadcast))
-				O.show_message(msg, m_type)
-			if (!(user in viewers(broadcast)))
-				user.show_message(msg, emote_type)
+	
+	for(var/mob/M in dead_mob_list)
+		if (!M.client)
+			continue //skip monkeys and leavers
+		if(findtext(message," snores.")) //Because we have so many sleeping people.
```

Comment:
what is this shitcode? and why is inside the loop?

Reply:
It's from the old emote code
I agree it's completly unnecessary

>and why is inside the loop?
You'd want to display snoring to people in view but not to ghosts ?